sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java

```diff
@@ -809,4 +808,13 @@ public static SqlQuerySpec getOfferQuerySpecFromResourceId(CosmosAsyncContainer
     public static CosmosAsyncContainer getControlContainerFromThroughputGlobalControlConfig(GlobalThroughputControlConfig globalControlConfig) {
         return globalControlConfig.getControlContainer();
     }
+
+    @Warning(value = INTERNAL_USE_ONLY_WARNING)
+    public static RetryContext getRetryContext(CosmosDiagnostics cosmosDiagnostics) {
+        if(cosmosDiagnostics != null && cosmosDiagnostics.clientSideRequestStatistics() != null) {
```

Comment:
didn't you say in the comment above that CosmosDiagnostics.clientSideRequestStatistics() gets always craeted for every request - so why the null check here?

Reply:
removed the null from here

Reply:
Correcting again , we need this null check as this code is called for gateway calls as well which can have non document resources, so it was here for that purpose